index.html

```diff
@@ -151,60 +151,110 @@ <h2>
-        <li>If <a>this</a>'s <a>relevant global object</a>'s <a>associated
-        <code>Document</code></a> is not <a>allowed to use</a> the
-        "<code>battery</code>" feature, then reject <a>this</a>'s <a>battery
-        promise</a> with a {{"NotAllowedError"}} {{DOMException}}, and return
-        <a>this</a>'s <a>battery promise</a>.
-          <div class="note">
-            In other words, this step rejects if the <a>associated
-            <code>Document</code></a>'s <a>browsing context</a>'s <a>active
-            document</a>'s <a>origin</a> is not <a>same origin-domain</a> with
-            the <a>origin</a> of the <a>current settings object</a> of this
-            {{Navigator}} object, unless specifically allowed by the document's
-            permissions policy.
-          </div>
-        </li>
-        <li>If <a>this</a>'s <a>battery manager</a> is not null, return
-        <a>this</a>'s <a>battery promise</a>.
-        </li>
-        <li>Set <a>this</a>'s <a>battery manager</a> to the result of
-        <a>creating a new <code>BatteryManager</code> object</a> in
-        <a>this</a>'s <a>relevant realm</a>.
-        </li>
-        <li>
-          <a>Resolve</a> <a>this</a>'s <a>battery promise</a> with
-          <a>this</a>'s <a>battery manager</a>.
-        </li>
-        <li>Return <a>this</a>'s <a>battery promise</a>.
-        </li>
-      </ol>
+      <section>
+        <h3>
+          Internal slots
+        </h3>
+        <table class="simple">
+          <thead>
+            <tr>
+              <th>
+                Internal slot
+              </th>
+              <th>
+                Initial value
+              </th>
+              <th>
+                Description
+              </th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td>
+                <dfn>[[\BatteryPromise]]</dfn>
+              </td>
+              <td>
+                `null`
+              </td>
+              <td>
+                A {{Promise}} returned by calls to {{Navigator/getBattery()}}.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                <dfn>[[\BatteryManager]]</dfn>
+              </td>
+              <td>
+                `null`
+              </td>
+              <td>
+                The <a>BatteryManager</a> instance associated with a given
+                {{Navigator}} after it has been created via
+                {{Navigator/getBattery()}}.
+              </td>
+            </tr>
+          </tbody>
+        </table>
+      </section>
+      <section>
+        <h3>
+          The `getBattery()` method
+        </h3>
+        <p data-tests=
+        "battery-promise-window.https.html, battery-promise.https.html">
+          The <dfn>getBattery()</dfn> method, when invoked, MUST run the
+          following steps:
+        </p>
+        <ol class="algorithm">
+          <li>If <a>this</a>'s {{Navigator/[[BatteryPromise]]}} is `null`, then
```

Comment:
I think you can modernize further by using this form throughout:

```suggestion
          <li>If [=this=].{{Navigator/[[BatteryPromise]]}} is `null`, then
```

Reply:
Good point. I've switched from `<a>foo</a>` to `[=foo=]` in the entire algorithm.